src/python/pants/backend/jvm/tasks/classpath_products.py

```diff
@@ -434,7 +435,8 @@ def _filter_by_excludes(self, classpath_target_tuples, root_targets):
     # set of targets was included here, their closure must be included.
     closure = BuildGraph.closure(root_targets, bfs=True)
     excludes = self._excludes.get_for_targets(closure)
-    return filter(_not_excluded_filter(excludes), classpath_target_tuples)
+    return [target_tuple for target_tuple in classpath_target_tuples
```

Comment:
This list comp is a little weird. `_not_excluded_filter(excludes)` returns a lambda, to which we pass the individual item `target_tuple`, hence the double parentheses. 

I tested that it works (tests were failing until I got the logic right).